source/Octopus.Client/Repositories/Async/ProjectRepository.cs

```diff
@@ -164,5 +167,29 @@ public Task<VersionControlBranchResource> GetVersionControlledBranch(ProjectReso
             var response = await client.Post<ConvertProjectToVersionControlledCommand,ConvertProjectToVersionControlledResponse>(url, payload);
             return response;
         }
+
+        public async Task<ResourceCollection<ChannelResource>> GetChannels(ProjectResource projectResource, string gitRef)
+        {
+            projectResource.EnsureVersionControlled();
+            VersionControlBranchResource branch = await GetVersionControlledBranch(projectResource, gitRef);
+            return await client.List<ChannelResource>(branch.Link("Channels"));
+        }
+
+        public async Task<IReadOnlyList<ChannelResource>> GetAllChannels(ProjectResource projectResource, string gitRef)
+        {
+            projectResource.EnsureVersionControlled();
+            VersionControlBranchResource branch = await GetVersionControlledBranch(projectResource, gitRef);
+            return await client.ListAll<ChannelResource>(branch.Link("Channels"));
+        }
+
+        public async Task<ChannelResource> GetChannel(ProjectResource projectResource, string gitRef, string idOrName)
+        {
+            projectResource.EnsureVersionControlled();
+
+            VersionControlBranchResource branch = await GetVersionControlledBranch(projectResource, gitRef);
+            var url = $"{branch.Link("Channels")}/{idOrName}";
+
+            return await client.Get<ChannelResource>(url);
+        }
```

Comment:
Each of these will send two requests. One to get the list of branches, and one to get the channel from the URL specified in the `VersionControlBranchResource`. Ideally, we could just sub the provided `gitRef` into the URL, though it doesn't look like we have that kind of capability (Unless I'm just blind, in which case, please let me know)

I've left these under the Beta repository so we can come back and change it later without too much fuss.

Same goes for the non-async repository.